### ISSUE TITLE
missing default for host.v1 config CONNECT TIMEOUT

### ISSUE THREAD

**Issue by qrkourier** [openziti/ziti-console#230]
In ziti-console 2.9.0, it's necessary to provide a valid value for the 'CONNECT TIMEOUT' field that's presented when attaching a host.v1 config to a service in the create service form. This should default to a sane value, and not add unnecessary friction to creating a service.

**Comment by qrkourier** [comment]
Additionally, the input validation on this field is not correct. I was able to enter a string `3s`, which triggers this router error.

```
[1245.361]   ERROR ziti/tunnel/intercept.(*ServiceListener).addService: {serviceId=[2xBkybHv13khwpwXwxTpnl] serviceName=[zubnet] error=[unable to decode service config structure: 1 error(s) decoding:

* 'ListenOptions.ConnectTimeout' expected type 'time.Duration', got unconvertible type 'string', value: '3s']} service is hostable but unable to decode server config of type host.v1
```

**Comment by qrkourier** [comment]
Plot twist: the `3s` form is actually _required_ by input validation despite being rejected by router v0.31.0, and I can neither create nor update a service with only an integer.

**Comment by qrkourier** [comment]
I confirmed this API property (`listenOptions.connectTimeout`) does expect the format required by the console input field in the create/update config form (e.g., `3s`), so the it seems the router's tunneler in host mode isn't able to parse this host.v1 config, and instead reports the error above (`got unconvertible type 'string'`).

```json
{
  "allowedAddresses": [
    "10.244.0.0/16"
  ],
  "allowedPortRanges": [
    {
      "high": 65535,
      "low": 1
    }
  ],
  "allowedProtocols": [
    "tcp",
    "udp"
  ],
  "forwardAddress": true,
  "forwardPort": true,
  "forwardProtocol": true,
  "listenOptions": {
    "bindUsingEdgeIdentity": false,
    "connectTimeout": "3s",
    "identity": "",
    "precedence": "default"
  }
}
```

**Comment by qrkourier** [comment]
@plorenz I confirmed that router 0.32.2 still can't parse that apparently-valid host.v1 config :point_up_2:

**Comment by qrkourier** [comment]
@rgallettonf I upgraded my console to 3.0.3 to check whether the missing default might have been corrected. I clicked this host.v1 config in the list and tried immediately to save it as-is. I got a different error this time: 

```json
{
    "cause": {
        "field": "proxy",
        "reason": "proxy is invalid: proxy: address is required",
        "value": "map[type:http]"
    },
    "code": "COULD_NOT_VALIDATE",
    "message": "The supplied request contains an invalid document or no valid accept content were available, see cause",
    "requestId": "95UrVrtj-"
}
```

I don't have a proxy, i.e. HTTP CONNECT forward proxy, in this case, so it's an irrelevant field and shouldn't be required. Both Ziti Controller and Router are 0.32.2 (latest), and the config entity was created with Controller and Router 0.31.0.

**Comment by qrkourier** [comment]
I worked around this by setting host.v1 config property `"listenOptions": {}` .

Ryan, I couldn't reproduce the two console errors with entities that were created with the latest CLI or latest console, so I believe the only remaining console problem is handling config entities that were created with a previous console version where input validation or the way the request object was constructed were broken. 

Specifically, it was previously necessary to set a connect timeout value like this because the config entity could not be created with a null value due to input validation rules. 

```json
  "listenOptions": {
    "bindUsingEdgeIdentity": false,
    "connectTimeout": "3s",
    "identity": "",
    "precedence": "default"
  }
```

The underlying problem that I still see in the latest console is that the shape of the request object as JSON is immediately updated when an empty field is focused with a single left click, thereby changing the set of required fields, but the shape of the request is not reverted if the user then declines to actually type any value. 

For example, if I happen to the type of user that clicks in to a field I'm looking at while I'm studying the form to understand its fields, then I may inadvertently commit to typing something in that field, though I don't necessarily have any understanding of or intention of using that field.

I think the shape of the request should be reverted to use the API's defaults when no values are provided for that section of the entity, so it will correctly reflect the user's intention.

```json
  "listenOptions": {}
```

In summary, I was forced to provide a connect timeout because I'd carelessly clicked in a field I was looking at but didn't intend to fill, which made that field required when I didn't want to use it or understand its implications. That led to the discovery that the config was invalid if only that value was supplied, drawing attention to a mismatch between console input validation and the router's (ER/T in host mode) expectations for the shape of the host.v1 config.

**Comment by plorenz** [comment]
Related issue: openziti/ziti#873
